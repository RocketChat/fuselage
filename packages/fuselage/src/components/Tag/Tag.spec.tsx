import React from 'react';
import ReactDOM from 'react-dom';

import { Tag } from '.';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Tag />, div);
  ReactDOM.unmountComponentAtNode(div);
});
