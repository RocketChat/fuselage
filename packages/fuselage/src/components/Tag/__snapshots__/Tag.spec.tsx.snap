// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Tag Rendering] renders _Danger without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-8i0p31"
    >
      <span
        class="rcx-box rcx-box--full rcx-tag rcx-tag--danger"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </span>
    </div>
  </div>
</body>
`;

exports[`[Tag Rendering] renders _Primary without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-8i0p31"
    >
      <span
        class="rcx-box rcx-box--full rcx-tag rcx-tag--primary"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </span>
    </div>
  </div>
</body>
`;

exports[`[Tag Rendering] renders _SecondaryDanger without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-8i0p31"
    >
      <span
        class="rcx-box rcx-box--full rcx-tag rcx-tag--secondary-danger"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </span>
    </div>
  </div>
</body>
`;

exports[`[Tag Rendering] renders AsLink without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-8i0p31"
    >
      <a
        class="rcx-box rcx-box--full rcx-tag rcx-tag--clickable"
        href="#"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </a>
    </div>
  </div>
</body>
`;

exports[`[Tag Rendering] renders Clickable without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-8i0p31"
    >
      <span
        class="rcx-box rcx-box--full rcx-tag rcx-tag--clickable"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </span>
    </div>
  </div>
</body>
`;

exports[`[Tag Rendering] renders Default without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-8i0p31"
    >
      <span
        class="rcx-box rcx-box--full rcx-tag "
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </span>
    </div>
  </div>
</body>
`;

exports[`[Tag Rendering] renders Disabled without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-8i0p31"
    >
      <span
        class="rcx-box rcx-box--full rcx-tag rcx-tag--disabled"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </span>
    </div>
  </div>
</body>
`;

exports[`[Tag Rendering] renders Featured without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-8i0p31"
    >
      <span
        class="rcx-box rcx-box--full rcx-tag rcx-tag--featured"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </span>
    </div>
  </div>
</body>
`;

exports[`[Tag Rendering] renders Large without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-8i0p31"
    >
      <span
        class="rcx-box rcx-box--full rcx-tag rcx-tag--large"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </span>
    </div>
  </div>
</body>
`;

exports[`[Tag Rendering] renders Medium without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-8i0p31"
    >
      <span
        class="rcx-box rcx-box--full rcx-tag rcx-tag--medium"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </span>
    </div>
  </div>
</body>
`;

exports[`[Tag Rendering] renders Secondary without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-8i0p31"
    >
      <span
        class="rcx-box rcx-box--full rcx-tag rcx-tag--secondary"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </span>
    </div>
  </div>
</body>
`;

exports[`[Tag Rendering] renders Variants without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-button-group rcx-button-group--align-start"
      role="group"
    >
      <span
        class="rcx-box rcx-box--full rcx-tag  rcx-button-group__item"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </span>
      <span
        class="rcx-box rcx-box--full rcx-tag rcx-tag--primary rcx-button-group__item"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </span>
      <span
        class="rcx-box rcx-box--full rcx-tag rcx-tag--danger rcx-button-group__item"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </span>
      <span
        class="rcx-box rcx-box--full rcx-tag rcx-tag--warning rcx-button-group__item"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </span>
      <span
        class="rcx-box rcx-box--full rcx-tag rcx-tag--featured rcx-button-group__item"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </span>
    </div>
  </div>
</body>
`;

exports[`[Tag Rendering] renders VariantsAsLinks without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-button-group rcx-button-group--align-start"
      role="group"
    >
      <a
        class="rcx-box rcx-box--full rcx-tag rcx-tag--clickable rcx-button-group__item"
        href="#"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </a>
      <a
        class="rcx-box rcx-box--full rcx-tag rcx-tag--primary rcx-tag--clickable rcx-button-group__item"
        href="#"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </a>
      <a
        class="rcx-box rcx-box--full rcx-tag rcx-tag--danger rcx-tag--clickable rcx-button-group__item"
        href="#"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </a>
      <a
        class="rcx-box rcx-box--full rcx-tag rcx-tag--warning rcx-tag--clickable rcx-button-group__item"
        href="#"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </a>
      <a
        class="rcx-box rcx-box--full rcx-tag rcx-tag--featured rcx-tag--clickable rcx-button-group__item"
        href="#"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </a>
    </div>
  </div>
</body>
`;

exports[`[Tag Rendering] renders Warning without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-8i0p31"
    >
      <span
        class="rcx-box rcx-box--full rcx-tag rcx-tag--warning"
      >
        <span
          class="rcx-tag__inner"
        >
          john.doe
        </span>
      </span>
    </div>
  </div>
</body>
`;

exports[`[Tag Rendering] renders WithIcon without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-8i0p31"
    >
      <span
        class="rcx-box rcx-box--full rcx-tag "
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-team-lock rcx-icon rcx-css-nbjs1j"
        >
          î‚¿
        </i>
        <span
          class="rcx-tag__inner"
        >
          Team
        </span>
      </span>
    </div>
  </div>
</body>
`;
